os/drivers/serial/serial.c

```diff
@@ -125,6 +132,87 @@ static const struct file_operations g_serialops = {
  * Private Functions
  ************************************************************************************/
 
+#ifdef CONFIG_DYNAMIC_LOGS_STORING
+struct log_net_chunk_s *get_log_net_head(void) {
+	return log_net_head;
+}
+
+static int read_counter = 0;
+
+void read_log_net(void) {
+        struct log_net_chunk_s *next_chunk = log_net_head;
+	struct mm_heap_s *myheap = mm_get_heap(log_net_head); /* get heap details using the head of the chunks linked list */
+
+                if (!myheap) {
+                        return; /* how to proceed if myheap is NULL? */
+                }
+
+                size_t free_size = myheap->mm_heapsize - myheap->total_alloc_size; /* get the free size */
+        lldbg("Read output %dth time!! ---> starts here <--- log size : %d, free heap : %d\n", read_counter, log_net_size, free_size);
```

Comment:
Please use `printf` instead of `lldbg`.
Because `lldbg` can mix the other logs, so we recommend to use it when assert case.